src/main/java/org/skriptlang/skript/bukkit/fishing/elements/ExprEntityBucket.java

```diff
@@ -0,0 +1,64 @@
+package org.skriptlang.skript.bukkit.fishing.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerBucketEntityEvent;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Entity Bucket")
+@Description("Gets the bucket that the entity will be put into such as 'puffer fish bucket'.")
+@Examples({
+	"on bucket capture entity:",
+		"\tif future item bucket is salmon bucket:",
+			"\t\tsend \"Congratulations you now have a salmon bucket!\" to player"
+})
+@Events("Bucket Catch Entity")
+@Since("INSERT VERSION")
+public class ExprEntityBucket extends SimpleExpression<ItemStack> {
+
+	static {
+		Skript.registerExpression(ExprEntityBucket.class, ItemStack.class, ExpressionType.EVENT,
+			"[the] (future|resulting) [event][-| ]item [bucket]");
```

Comment:
i don't think this should pretend to be an event-value
Can't you just register a future item event value for the bucket event?